### ISSUE TITLE
How to push home-manager generation to cachix?

### ISSUE THREAD

**Issue by samuela** [cachix/cachix#541]
This is a question that really feels like it must have been asked before, but I have been surprised to come up empty handed with my googling thus far... How does one push a home-manager generation, eg everything necessary for `home-manager switch`, to a cachix cache?

https://github.com/cachix/cachix/issues/259 covers configuring cachix from within home-manager, but what I'm seeking is a solution to not have to rebuild the world when running `home-manager switch`. [This is the page](https://docs.cachix.org/pushing) from the docs that I would expect to have the answer, but it does not mention home-manager.

Running `home-manager switch | cachix push mycache` does not work for me:
```
❯ home-manager switch | cachix push ploop
replacing old 'home-manager-path'
installing 'home-manager-path'

There are 95 unread and relevant news items.
Read them by running the command "home-manager news".

cachix: CppStdException e "\ESC[31;1merror:\ESC[0m getting status of '\ESC[35;1m/home/sam.ainsworth/external-dev/nixpkgs/Starting\ESC[0m': \ESC[35;1mNo such file or directory\ESC[0m"(Just "nix::SysError")
```

So, how can one push their home-manager builds to cachix?

**Comment by domenkozar** [comment]
You can do `cachix watch-exec ploop home-manager switch`.

**Comment by samuela** [comment]
> There's an option in the cache settings to use cache.nixos.org as an upstream cache. It's enabled by default. Store paths available in cache.nixos.org won't be pushed to save space. Does that account for what you're seeing?

Yes, perhaps I should clarify! It's possible that could be the cause, although I'm not entirely certain. In my home-manager config I have `nvtop` which depends on `cudatoolkit`. Due to being unfree, `cudatoolkit` and its dependencies are not cached in `cache.nixos.org` and so both packages get built locally when running a clean `home-manager switch`. IIRC `cudatoolkit` was absent from the output of my `cachix push`, which led me to conclude that perhaps it was missing. 

I could be mistaken here... I'm guessing that cachix is doing it right and i'm goofing something up! Perhaps I misread or something is cached elsewhere that I wasn't expecting

**Comment by domenkozar** [comment]
It should create `result` symlink so that you can do `cachix push mycache ./result`.
